connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/rbac/RBACUsersProcessor.java

```diff
@@ -0,0 +1,77 @@
+package com.redhat.cloud.notifications.connector.email.processors.rbac;
+
+import com.redhat.cloud.notifications.connector.email.constants.ExchangeProperty;
+import com.redhat.cloud.notifications.connector.email.model.settings.RecipientSettings;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashSet;
+import java.util.Set;
+
+@ApplicationScoped
+public class RBACUsersProcessor implements Processor {
+    /**
+     * Processes the incoming payload from the RBAC service into a filtered
+     * set of usernames. In the case that the users are part of an RBAC group
+     * it is looked if only admins are allowed and if the user is active before
+     * adding it to the set.
+     * @param exchange the exchange of the pipeline.
+     */
+    @Override
+    public void process(final Exchange exchange) {
+        final String responseBody = exchange.getIn().getBody(String.class);
+        final JsonObject responseBodyJson = new JsonObject(responseBody);
+
+        // Extract the data we need to process.
+        final JsonArray data = responseBodyJson.getJsonArray("data");
+
+        // Get ready the set of usernames that we will transform from the
+        // incoming payload. If multiple pages are fetched, this might get
+        // called multiple times, that is why we need to be safe.
+        Set<String> usernames = exchange.getProperty(ExchangeProperty.USERNAMES, Set.class);
+        if (usernames == null) {
+            usernames = new HashSet<>();
+            exchange.setProperty(ExchangeProperty.USERNAMES, usernames);
+        }
```

Comment:
Can this really be executed since `DispatcherProcessor` creates the `HashSet` instance?

Reply:
Nope! Removed!